src/server/nuclearnet/tests/fake_nuclearnet_client.tests.ts

```diff
@@ -0,0 +1,192 @@
+import { FakeNUClearNetClient } from '../fake_nuclearnet_client'
+import { FakeNUClearNetServer } from '../fake_nuclearnet_server'
+
+describe('FakeNUClearNetClient', () => {
```

Comment:
If you really want to ensure that you're simulating NUClear properly there are a few other things you should put in the tests/code as they may have an influence on the behaviour of the system.
- when connecting a single client will connect from an address/port. This port is assigned by the OS and if you disconnect and reconnect, you'll often have a different port.
- the address may be either an ipv4 or ipv6 address, so make sure you don't assume it's a dotted decimal address.